HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,44 @@
+def bfs(char_pairs, start_char, sc):
+    """
+    Given a graph RDD and a starting character name, performs a breadth-first search and
+    returns an RDD of all nodes encountered.
+    """
+    
+    def start_dist(k,v):
+    	" Changes the distance for the starting character to zero. "
+    	if k == start_char:
+        	return (k,0)
+    	else:
+        	return (k,v)
+    
+    def update_dist(v):
+		" Updates nodes' distance from the starting node and uses an accumulator to check if any distances are updated. "
+		if v[1] < 0:
+			accum.add(1)
+			return (v[0], counter)
+		return v
+        
+    accum = sc.accumulator(0)
+    dist = char_pairs.keys().distinct().map(lambda v: (v,-1))
+    dist = dist.map(lambda (k,v): start_dist(k,v))
```

Comment:
might be a little more clear if you combined these two lines into one map of an "initialize_dist" function